### ISSUE TITLE
Add PyPA DB source

### ISSUE THREAD

**Issue by westonsteimel** [anchore/grype#531]
Add the ability to match python packages against vulnerabilities listed in https://github.com/pypa/advisory-db

**Comment by Percivalll** [comment]
Yes, this is useful for matching Python package vulnerabilities, especially those without GHSA or CVE IDs. For example, the malicious versions of LiteLLM:
https://github.com/pypa/advisory-database/blob/b0f7a727494c977b29c998bc9199de5891f8f302/vulns/litellm/PYSEC-2026-2.yaml
Or should we discuss this feature request in the [vunnel repo](https://github.com/anchore/vunnel)?

**Comment by willmurphyscode** [comment]
@Percivalll discussing here is fine. There will be pull requests to both repos probably (adding a new vunnel provider and updating database transformers and matchers in Grype to account for the new data).

It looks to me like this is an OSV yaml dataset, similar to R-consortium's (see https://github.com/anchore/vunnel/pull/1043 )? And we could just get an osv.dev zip file?

**Comment by westonsteimel** [comment]
You can also just get a zip from the repo directly though: https://github.com/pypa/advisory-database/archive/refs/heads/main.zip

**Comment by westonsteimel** [comment]
The difficulty with integrating these datasets at the moment with grype is the overlap that happens and trying to figure out a way to dedup results so it doesn't overwhelm people.  Because a large majority of the data will overlap with GitHub Advisories, so we'd eventually like to just show one result for the same match rather than multiple.  There is also the drift that happens, often the upstream source only has the initial data, but then edits get made to GitHub advisories after the fact (because the Advisory edit UI is easy to use probably more than anything else), but those same changes don't make it back to the corresponding sources

**Comment by westonsteimel** [comment]
That doesn't mean we shouldn't bring it in, just that we need to consider these issues

**Comment by westonsteimel** [comment]
I am working in the background on some of these things by working on a new more unified dataset, but that is still a ways from being useful, but some of the things I plan to do here is consider the update timestamps from the sources and deduplicate that way.  So if the GHSA affected package data has been modified more recently than a PyPA entry for the same package, take the GHSA data for that package instance and just take the Pypa identifier as an alias, or vice versa.

**Comment by westonsteimel** [comment]
The PyPA data also allows adding affected function call information which will be useful to capture and potentially utilise in future (similar to the ask for Go vuln db data), but last I looked there were very few records that actually populate that information yet on the PyPA side

**Comment by westonsteimel** [comment]
I haven't looked at it for a few months though
